common/src/operator.rs

```diff
@@ -45,53 +46,57 @@ pub enum Operator {
   Assign,
 }
 
-impl Operator {
-  pub fn from(op: &str) -> Option<Operator> {
-    match op {
-      "&&=" => Some(Operator::LogAndAssign),
-      "||=" => Some(Operator::LogOrAssign),
-      "&&" => Some(Operator::LogAnd),
-      "||" => Some(Operator::LogOr),
-      "!" => Some(Operator::LogNot),
-
-      "+=" => Some(Operator::AddAssign),
-      "-=" => Some(Operator::SubAssign),
-      "*=" => Some(Operator::MulAssign),
-      "/=" => Some(Operator::DivAssign),
-      "%=" => Some(Operator::ModAssign),
-      "**=" => Some(Operator::ExpAssign),
-      "+" => Some(Operator::Add),
-      "-" => Some(Operator::Sub),
-      "*" => Some(Operator::Mul),
-      "/" => Some(Operator::Div),
-      "%" => Some(Operator::Mod),
-      "**" => Some(Operator::Exp),
-
-      "<<=" => Some(Operator::BitLeftShiftAssign),
-      ">>=" => Some(Operator::BitRightShiftAssign),
-      "<<" => Some(Operator::BitLeftShift),
-      ">>" => Some(Operator::BitRightShift),
-      "&=" => Some(Operator::BitAndAssign),
-      "|=" => Some(Operator::BitOrAssign),
-      "^=" => Some(Operator::BitXorAssign),
-      "&" => Some(Operator::BitAnd),
-      "|" => Some(Operator::BitOr),
-      "^" => Some(Operator::BitXor),
-      "~" => Some(Operator::BitNot),
-
-      "==" => Some(Operator::Eq),
-      "!=" => Some(Operator::NotEq),
-      "<=" => Some(Operator::LessThanOrEq),
-      ">=" => Some(Operator::GreaterThanOrEq),
-      "<" => Some(Operator::LessThan),
-      ">" => Some(Operator::GreaterThan),
-
-      "=" => Some(Operator::Assign),
-
-      _ => None,
+impl TryFrom<&str> for Operator {
+  type Error = ();
+
+  fn try_from(operator: &str) -> Result<Self, Self::Error> {
+    match operator {
+      "&&=" => Ok(Operator::LogAndAssign),
+      "||=" => Ok(Operator::LogOrAssign),
+      "&&" => Ok(Operator::LogAnd),
+      "||" => Ok(Operator::LogOr),
+      "!" => Ok(Operator::LogNot),
```

Comment:
All logical operators have to be evaluated to 0 or 1 and then bitwise equivelant should be used